domain/secretbackend/service/watcher.go

```diff
@@ -14,25 +14,25 @@ import (
 	"github.com/juju/juju/internal/errors"
 )
 
-type secretBackendRotateWatcher struct {
+type secretBackendRotateWatcher[T any] struct {
 	catacomb      catacomb.Catacomb
 	sourceWatcher watcher.StringsWatcher
 	logger        logger.Logger
 
-	processChanges func(ctx context.Context, backendIDs ...string) ([]watcher.SecretBackendRotateChange, error)
+	processChanges func(ctx context.Context, backendIDs ...string) ([]T, error)
 
-	out chan []watcher.SecretBackendRotateChange
+	out chan []T
 }
 
-func newSecretBackendRotateWatcher(
+func newSecretBackendRotateWatcher[T any](
```

Comment:
How does this compile without a change to 
`return newSecretBackendRotateWatcher(w, s.logger, s.st.GetSecretBackendRotateChanges)`
https://github.com/juju/juju/blob/8ffc0b8b3249954069223f25f4c26ad428b7ff83/domain/secretbackend/service/service.go#L746

What's the reason for the change here?

Reply:
It's inferred from the `GetSecretBackendRotateChanges`.